process/track/sideChainShardBlockTrack.go

```diff
@@ -0,0 +1,51 @@
+package track
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+type sideChainShardBlockTrack struct {
+	*shardBlockTrack
+}
+
+// NewSideChainShardBlockTrack creates an object for tracking the received shard blocks
+func NewSideChainShardBlockTrack(shardBlockTrack *shardBlockTrack) (*sideChainShardBlockTrack, error) {
+	if shardBlockTrack == nil {
+		return nil, process.ErrNilBlockTracker
+	}
+
+	sideChainBlockTrack := &sideChainShardBlockTrack{
+		shardBlockTrack,
+	}
+
+	originalBlockProcessor, ok := sideChainBlockTrack.blockProcessor.(*blockProcessor)
+	if !ok {
+		return nil, process.ErrWrongTypeAssertion
+	}
+
+	newBlockProcessor, err := NewSideChainBlockProcessor(originalBlockProcessor)
+	if err != nil {
+		return nil, err
+	}
+
+	sideChainBlockTrack.blockProcessor = newBlockProcessor
+	return sideChainBlockTrack, nil
+}
+
+// ComputeLongestSelfChain computes the longest chain from self shard
+func (scsbt *sideChainShardBlockTrack) ComputeLongestSelfChain() (data.HeaderHandler, []byte, []data.HeaderHandler, [][]byte) {
```

Comment:
this will work as expected if it is only called from exterior. Also the next one

Reply:
Fixed